rostok/graph_generators/environments/design_environment.py

```diff
@@ -0,0 +1,775 @@
+from collections import defaultdict
+from copy import deepcopy
+import pickle
+from abc import ABC, abstractmethod
+from typing import Any, Union, TypeAlias
+import os
+from datetime import datetime
+
+import numpy as np
+import networkx as nx
+import matplotlib.pyplot as plt
+
+from rostok.graph_grammar.rule_vocabulary import RuleVocabulary
+from rostok.graph_grammar.node import GraphGrammar
+from rostok.trajectory_optimizer.control_optimizer import GraphRewardCalculator
+
+STATESTYPE = Union[int, str]
+
+StepType: TypeAlias = tuple[STATESTYPE, float, bool, bool]
+TransitionFunctionType: TypeAlias = dict[tuple[STATESTYPE, int], tuple[STATESTYPE, float, bool]]
+
+class Environment(ABC):
```

Comment:
Это ведь не общая форма для Environment, тут сразу предполагается, что reward бывает только у терминальных состояний. Может надо как-то по-другому назвать? Вдруг у нас появится система, когда нетерминальные стейты имеют реворды. Или тогда изменится понятие о терминальнои и нетерминальном?

Reply:
Можно всегда переопределить метод. Поменяю название тогда 

```suggestion
class EnvironmentTerminalReward(ABC):
```